src/main/scala/is/hail/variant/Genotype.scala

```diff
@@ -903,14 +822,29 @@ object Genotype {
       (1, Gen.const(null)))
   }
 
+
+  def genGenericDosageGenotype(v: Variant): Gen[Annotation] = {
+    val nAlleles = v.nAlleles
+    val nGenotypes = triangle(nAlleles)
+    val gg = for (gp <- Gen.option(Gen.partition(nGenotypes, 32768))) yield {
+      val gt = gp.flatMap(gtFromLinear)
+      Row(
+        gt.orNull,
+        gp.map(gpx => gpx.map(p => p.toDouble / 32768): IndexedSeq[Double]).orNull)
+    }
+    Gen.frequency(
+      (100, gg),
+      (1, Gen.const(null)))
+  }
+
   def genVariantGenotype: Gen[(Variant, Genotype)] =
     for (v <- Variant.gen;
-      g <- Gen.oneOfGen(genExtreme(v), genRealistic(v), genDosageGenotype(v)))
+      g <- Gen.oneOfGen(genExtreme(v), genRealistic(v)))
```

Comment:
I was debating whether to say you should include genGenericDosageGenotype here. I'm assuming you took it out because it's tested in `VSM.dosage`.

Reply:
I took it out because `genGenericDosageGenotype` returns an Annotation, but this is a `Gen` of `Genotype`.  When everything goes generic, this can go back in.